mapr/datadog_checks/mapr/mapr.py

```diff
@@ -0,0 +1,153 @@
+    import confluent_kafka as ck
+except ImportError:
+    ck = None
+
+
+DEFAULT_STREAM_PATH = "/var/mapr/mapr.monitoring/metricstreams"
+METRICS_SUBMITTED_METRIC_NAME = "mapr.metrics.submitted"
+SERVICE_CHECK = "mapr.can_connect"
+
+"""
+All mapr metrics go through a Stream topic (similar to a Kafka topic) and are consumed by OpenTSDB, a timeserie
+database.
+All the metrics are distributed over multiple topics, one for each host. This way, the check instance can subscribe
+to the topic relevant to the current host and consume everything.
+
+Note: The MapR documentation https://mapr.com/docs/61/AdministratorGuide/spyglass-on-streams.html states that
+there is one topic per host per metric name. This is wrong starting with 6.1+, there is only one topic per host now.
+To support older versions of MapR, the check should be updated to subscribe those multiple topics.
+"""
+
+
+class MaprCheck(AgentCheck):
+    def __init__(self, name, init_config, instances):
+        super(MaprCheck, self).__init__(name, init_config, instances)
+        self._conn = None
+        self.hostname = self.instance.get('hostname', get_fqdn())
+        self.streams_count = self.instance.get('streams_count', 2)
+        self.topic_path = "{stream_path}/{stream_id}:{topic_name}".format(
+            stream_path=self.instance.get('stream_path', DEFAULT_STREAM_PATH),
+            stream_id=get_stream_id_for_topic(self.hostname, rng=self.streams_count),
+            topic_name=self.hostname,
+        )
+        self.allowed_metrics = [re.compile(w) for w in self.instance.get('metric_whitelist', [])]
+        self.base_tags = self.instance.get('tags', [])
+        self.is_first_check_run = True
+
+        auth_ticket = self.instance.get('ticket_location')
+        if auth_ticket:
+            os.environ['MAPR_TICKETFILE_LOCATION'] = auth_ticket
+        elif not os.environ.get('MAPR_TICKETFILE_LOCATION'):
+            self.log.info("MAPR_TICKETFILE_LOCATION environment variable not set, this may cause authentication issues")
+
+    def check(self, _):
+        if ck is None:
+            raise CheckException(
+                "confluent_kafka was not imported correctly, make sure the library is installed and that you've "
+                "set LD_LIBRARY_PATH correctly. Please refer to datadog documentation for more details."
+            )
+
+        try:
+            conn = self.get_connection()
+        except Exception:
+            self.service_check(
+                SERVICE_CHECK, AgentCheck.CRITICAL, self.base_tags + ['topic_path:{}'.format(self.topic_path)]
+            )
+            raise
+
+        submitted_metrics_count = 0
+
+        while True:
+            msg = conn.poll(timeout=0.5)
+            if msg is None:
+                # Timed out, no more messages
+                break
+
+            if msg.error() is None:
+                # Metric received
+                try:
+                    metric = json.loads(ensure_unicode(msg.value()))[0]
```

Comment:
Does `poll` always return one metric result result ? Any need to loop values instead of `[0]` ?

Reply:
Metrics are always produced one-by-one. If one day something contains more than one element, it probably means something completely different anyway. So no need to loop here